pkg/blobstore/cloud_blob_access.go

```diff
@@ -59,14 +61,25 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 	}
 	// In case of an error (e.g. network failure), we cancel before closing to
 	// request the write to be aborted.
-	if _, err = io.Copy(w, r); err != nil {
```

Comment:
It looks like this code predates the addition of `Buffer.IntoWriter()`. Maybe it would be best to replace this line by:

```go
if _, err := b.IntoWriter(w); err != nil {
```

That way this backend becomes agnostic of the way data is passed in. If it turns out this still causes OOM issues, then it must either be something wrong in pkg/blobstore/buffer, go-cloud, the AWS SDK, etc.

Reply:
I think this is the actual fix to this, I've made this change and am just testing that it does fix my issue. I guess this should teach me to read APIs more carefully!

Reply:
This didn't fix the issue, but the fact it didn't leads me to believe that the issue is not in the bb-storage source code. In any case, I think this is still a beneficial change.